instituto-brasileiro-de-informacao-em-ciencia-e-tecnologia-abnt.csl

```diff
@@ -381,15 +412,71 @@
       </else>
     </choose>
   </macro>
-  <macro name="archive">
-    <group>
-      <text variable="archive" prefix=" "/>
-    </group>
+  <macro name="publisher">
+    <choose>
+      <if variable="publisher-place publisher" match="any">
+        <group delimiter=": ">
+          <choose>
+            <if variable="publisher-place">
+              <text variable="publisher-place"/>
+            </if>
+            <else>
+              <text value="S. l." font-style="italic" prefix="[" suffix="]"/>
+            </else>
+          </choose>
+          <choose>
+            <if variable="publisher">
+              <text variable="publisher"/>
+            </if>
+            <else>
+              <text value="s. n." font-style="italic" prefix="[" suffix="]"/>
+            </else>
+          </choose>
+        </group>
+      </if>
+      <else>
+        <group delimiter=": ">
+          <text value="S. l." font-style="italic" prefix="["/>
+          <text value="s. n." font-style="italic" suffix="]"/>
+        </group>
+      </else>
+    </choose>
   </macro>
-  <macro name="number">
-    <group>
-      <text variable="number" prefix=", n. "/>
-    </group>
+  <macro name="recipient">
+    <names variable="recipient" delimiter=", ">
+      <label form="long" text-case="capitalize-first" suffix=": "/>
+      <name delimiter="; " sort-separator=" " delimiter-precedes-last="always">
+        <name-part name="given"/>
+        <name-part name="family"/>
+      </name>
+      <et-al font-style="italic"/>
+    </names>
+  </macro>
+  <macro name="title">
+    <choose>
+      <if type="article article-newspaper article-magazine article-journal entry-dictionary entry-encyclopedia chapter bill paper-conference webpage" match="any">
+        <text variable="title"/>
+      </if>
+      <else-if type="personal_communication" match="any">
+        <text variable="title" font-weight="bold" prefix="[" suffix="]"/>
+      </else-if>
+      <else-if type="song broadcast" match="any">
+        <text variable="title" text-case="uppercase"/>
+      </else-if>
+      <else>
+        <text variable="title" font-weight="bold"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="translator">
+    <text term="translator" form="verb-short" suffix=" "/>
```

Comment:
This should be a label under the 'names' element, not hard coded text. As is, the text always appears even if there is no translator.